### ISSUE TITLE
Don't leak all users and use invitations for channels

### ISSUE THREAD

**Issue by pavelzw** [prefix-dev/prefix-dev#25]
When clicking on `Add` in a channel, you can find a list of all users very quickly.
Is this intended?
![image](https://github.com/prefix-dev/prefix-dev/assets/29506042/d58226c7-6d7c-406c-a8b9-f540d4d904c8)

Also, when adding a user to a channel as a member, they are not asked whether they actually want to be a member of said channel 
 but get added directly 😅

**Comment by ruben-arts** [comment]
Yes we have talked about this. Currently that is by design. But maybe that needs to be questioned a little more ;)

**Comment by wolfv** [comment]
When we designed this we took inspiration from Github, which also "leaks" all users in this way.

I agree that it might be better to come up with an invitation system for adding channel members.
